cpu/stm32/periph/uart.c

```diff
@@ -116,10 +116,10 @@ static inline void uart_init_cts_pin(uart_t uart)
 
 static inline void uart_init_pins(uart_t uart, uart_rx_cb_t rx_cb)
 {
-     /* configure TX pin */
-    gpio_init(uart_config[uart].tx_pin, GPIO_OUT);
     /* set TX pin high to avoid garbage during further initialization */
     gpio_set(uart_config[uart].tx_pin);
+     /* configure TX pin after setting it to avoid a momentary low state */
+    gpio_init(uart_config[uart].tx_pin, GPIO_OUT);
```

Comment:
You might want to add a comment on why `gpio_set()` is called prior to the initialization and why this is safe.

The GPIO API does not guarantee that `gpio_set()` will do anything useful without prior initialization; I would consider such thing as "unspecified behavior". But as this code is in `cpu/stm32/`, it can rely on implementation details of the STM32 GPIO implementation. This should however be documented IMO, as this might cause issues in the future if the STM32 GPIO implementation is changed.